File: Solutions/1566. Detect Pattern of Length M Repeated K or More Times/1566. Detect Pattern of Length M Repeated K or More Times.java
class Solution {
    public boolean containsPattern(int[] arr, int m, int k) {
        int cnt=0;
        for(int i=0;i+m<arr.length;++i){
            if(arr[i]!=arr[i+m])
                cnt=0;
            cnt+=(arr[i]==arr[i+m])?1:0;
            if(cnt==(k-1)*m)
                return true;
        }
        return false;
    }
}

File: Solutions/1566. Detect Pattern of Length M Repeated K or More Times/1566. Detect Pattern of Length M Repeated K or More Times.go
func containsPattern(arr []int, m int, k int) bool {
    cnt:=0
    for i:=0;i+m<len(arr);i+=1{
        if arr[i]!=arr[i+m]{
            cnt=0
        }
        if arr[i]==arr[i+m]{
            cnt+=1
        }
        if cnt==(k-1)*m{
            return true
        }
    }
    return false
}